tests/evaluate/logging/utils.py

```diff
@@ -0,0 +1,230 @@
+    """
+    schema = _get_assessments_dataframe_schema()
+    return pd.read_json(path, orient="split", dtype=schema, convert_dates=False).replace(
+        pd.NA, None
+    )
+
+
+def _read_metrics_dataframe(path: str) -> pd.DataFrame:
+    """
+    Reads a metrics DataFrame from a file.
+
+    Args:
+        path (str): Path to the file.
+
+    Returns:
+        pd.DataFrame: The metrics DataFrame.
+    """
+    schema = _get_metrics_dataframe_schema()
+    return pd.read_json(path, orient="split", dtype=schema, convert_dates=False).replace(
+        pd.NA, None
+    )
+
+
+def _read_tags_dataframe(path: str) -> pd.DataFrame:
+    """
+    Reads a tags DataFrame from a file.
+
+    Args:
+        path (str): Path to the file.
+
+    Returns:
+        pd.DataFrame: The tags DataFrame.
+    """
+    schema = _get_tags_dataframe_schema()
+    return pd.read_json(path, orient="split", dtype=schema, convert_dates=False).replace(
+        pd.NA, None
+    )
+
+
+def _get_evaluation_from_dataframes(
+    *,
+    run_id: str,
+    evaluation_id: str,
+    evaluations_df: pd.DataFrame,
+    metrics_df: pd.DataFrame,
+    assessments_df: pd.DataFrame,
+    tags_df: pd.DataFrame,
+) -> EvaluationEntity:
+    """
+    Parses an Evaluation object with the specified evaluation ID from the specified DataFrames.
+    """
+    evaluation_row = evaluations_df[evaluations_df["evaluation_id"] == evaluation_id]
+    if evaluation_row.empty:
+        raise MlflowException(
+            f"The specified evaluation ID '{evaluation_id}' does not exist in the run '{run_id}'.",
+            error_code=RESOURCE_DOES_NOT_EXIST,
+        )
+
+    evaluations: List[EvaluationEntity] = _dataframes_to_evaluations(
+        evaluations_df=evaluation_row,
+        metrics_df=metrics_df,
+        assessments_df=assessments_df,
+        tags_df=tags_df,
+    )
+    if len(evaluations) != 1:
+        raise MlflowException(
+            f"Expected to find a single evaluation with ID '{evaluation_id}', but found "
+            f"{len(evaluations)} evaluations.",
+            error_code=INTERNAL_ERROR,
+        )
+
+    return evaluations[0]
+
+
+def _dataframes_to_evaluations(
+    evaluations_df: pd.DataFrame,
+    metrics_df: pd.DataFrame,
+    assessments_df: pd.DataFrame,
+    tags_df: pd.DataFrame,
+) -> List[EvaluationEntity]:
+    """
+    Converts four separate DataFrames (main evaluation data, metrics, assessments, and tags) back
+    into a list of Evaluation entities.
+
+    Args:
+        evaluations_df (pd.DataFrame): DataFrame with the main evaluation data
+            (excluding assessment and metrics).
+        metrics_df (pd.DataFrame): DataFrame with metrics.
+        assessments_df (pd.DataFrame): DataFrame with assessments.
+        tags_df (pd.DataFrame): DataFrame with tags.
+
+    Returns:
+        List[EvaluationEntity]: A list of Evaluation entities created from the DataFrames.
+    """
+
+    def group_by_evaluation_id(df: pd.DataFrame):
```

Comment:
can this function be moved in the top level?

Reply:
Why? it isn't used anywhere else

Reply:
It's just my habit to avoid defining function every time this function is called, avoid unintended variable shadowing, and improve readability. Not a blocker.

Reply:
Got it. I've moved it to the top level and fixed docstrings :)